RUFAS/time.py

```diff
@@ -15,41 +16,35 @@ def __init__(self):
         """
         This object is responsible for creating and tracking time in the simulation.
         """
+        self.config_data: Dict[str, str | int | bool] = im.get_data("config")
+        self.start_date: datetime = datetime.datetime.strptime(self.config_data["start_date"], "%Y:%j")
+        self.end_date: datetime = datetime.datetime.strptime(self.config_data["end_date"], "%Y:%j")
 
-        self._init_time_config()
-        self.calendar_year: int = self.start_year_int
-        self.year: int = 1  # current year
-        self.simulation_length = self._calc_sim_length()
+        self.current_date: datetime = self.start_date
+        self.simulation_length_days: int = (self.end_date - self.start_date).days
+        self.simulation_length_years: int = self.end_date.year - self.start_date.year + 1
 
-        # finds the first non-null day of the first year
-        for i in range(0, len(self.years[0])):
-            if self.years[0][i] is None:
-                continue
-            else:
-                self.day = self.years[0][i]
-                break
-
-        self.simulation_day = 0
-
-    def _init_time_config(self) -> None:
+    def advance(self) -> None:
         """
-        Initializes the time configuration for the instance by parsing the config data from InputManager pool.
+        Advances the time in the simulation by 1 day and automatically detects end of years.
```

Comment:
How does this method automatically detect the end of the year?

```suggestion
        Advances the time in the simulation by 1 day and automatically detects ends of years.
```